synapse/federation/federation_server.py

```diff
@@ -972,13 +972,21 @@ async def _process_incoming_pdus_in_room_inner(
         # the room, so instead of pulling the event out of the DB and parsing
         # the event we just pull out the next event ID and check if that matches.
         if latest_event is not None and latest_origin is not None:
-            (
-                next_origin,
-                next_event_id,
-            ) = await self.store.get_next_staged_event_id_for_room(room_id)
-            if next_origin != latest_origin or next_event_id != latest_event.event_id:
+            result = await self.store.get_next_staged_event_id_for_room(room_id)
+            if result is None:
                 latest_origin = None
                 latest_event = None
+            else:
+                (
+                    next_origin,
+                    next_event_id,
+                ) = result
```

Comment:
```suggestion
                next_origin, next_event_id  = result
```

Reply:
I'm surprised lint was happy with this. I think I assumed black would just reformat it. Oops!